### PR TITLE
regula falsi method in c++ and java

### LINKED ISSUE #1643: Implementation of regula falsi in c++ and java
I want to work on this issue for GSSoC 2020

### PR DESCRIPTION
**Pull request by yuktirapartiwar**
Fixes #1643 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Regula Falsi method in C++ and java
- 

#### Languages Used:
- c++
- java

#### Files Added:
- regula_falsi.cpp
- regula_falsi.java

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Regula_Falsi_Method/regula_falsi.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+}
+//
+//Output -
+//        Enter x1 =
+//        2
+//        Enter x2 =
+//        3
+//        Value of root is 2.0945182
```
Comment (jainaman224) [inline Regula_Falsi_Method/regula_falsi.java]: Add a single new line at the end of file.

File: Regula_Falsi_Method/regula_falsi.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            return;
+        }
+        float x3 = x1;
+        do{
+            x3 = (float)(x1 - (x2 - x1) / (f(x2) - f(x1)) * f(x1));
+            if(f(x3) * f(x1) < 0)
+                x2 = x3;
+            else
+                x1 = x3;
+        }while(abs(f(x3)) > 0.001);
```
Comment (jainaman224) [inline Regula_Falsi_Method/regula_falsi.java]: Add space after `}`.

File: Regula_Falsi_Method/regula_falsi.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        return (x * x * x - 2 * x - 5);
+    }
+    //Function to find root of f(x)
+    private static void regulaFalsi(float x1, float x2) {
+        if( f(x1) * f(x2) >= 0){
+            System.out.println("You have assumed incorrect x1 and x2");
+            return;
+        }
+        float x3 = x1;
+        do{
```
Comment (jainaman224) [inline Regula_Falsi_Method/regula_falsi.java]: Add space before `{`.

File: Regula_Falsi_Method/regula_falsi.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import static java.lang.StrictMath.abs;
+
+public class Main {
+    //Function to calculate f(x) where f(x) = x^3 - 2x - 5
+    static double f(float x){
+        return (x * x * x - 2 * x - 5);
+    }
+    //Function to find root of f(x)
+    private static void regulaFalsi(float x1, float x2) {
+        if( f(x1) * f(x2) >= 0){
```
Comment (jainaman224) [inline Regula_Falsi_Method/regula_falsi.java]: Add space before `(`, `{`.
Add space after `)`.

File: Regula_Falsi_Method/regula_falsi.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package com.company;
+//Regula Falsi method in java
+import java.util.Scanner;
+
+import static java.lang.StrictMath.abs;
+
+public class Main {
+    //Function to calculate f(x) where f(x) = x^3 - 2x - 5
```
Comment (jainaman224) [inline Regula_Falsi_Method/regula_falsi.java]: Add space after `//`.

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+int main() {
+    float x1, x2;
+    cout<<"Enter x1 = ";
+    cin>>x1;
+    cout<<"Enter x2 = ";
+    cin>>x2;
+    regulaFalsi(x1,x2);
+    
+    return 0;
+}
```
Comment (jainaman224) [inline Regula_Falsi_Method/regula_falsi.cpp]: Add a new line after this.

File: Regula_Falsi_Method/regula_falsi.cpp [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+            x2 = x3;
+        else
+            x1 = x3;
+    }while(abs(f(x3)) > 0.001);
+    cout<<"Value of root is " << x3;
+}
+
+int main() {
+    float x1, x2;
+    cout<<"Enter x1 = ";
```
Comment (jainaman224) [inline Regula_Falsi_Method/regula_falsi.cpp]: Add spaces around `<<`, `>>`.

**Comment by yuktirapartiwar** [comment]
> Do the changes throughout the file, wherever applicable.

All the changes have been completed.

**Comment by somya-kapoor** [comment]
Please squash the commits.

**Comment by yuktirapartiwar** [comment]
> Please squash the commits.

squashed the commits

**Comment by somya-kapoor** [comment]
Please create a new PR with Regula falsi method in java as in c++ it is already present.

**Pull request closed without merging**